client/components/checklist/task.js

```diff
@@ -106,25 +105,27 @@ class Task extends PureComponent {
 		return null;
 	}
 
-	onTaskClick = event => {
-		const { completed, id, onClick, onCollapsedClick } = this.props;
+	onTaskTitleClick = event => {
+		event.preventDefault();
+		const { id, onTaskTitleClick } = this.props;
 
-		// We presuppose expandability based on the presence of `onCollapsedClick`
-		if ( ! completed && this.isTaskCollapsed() && onCollapsedClick ) {
-			event.preventDefault();
-			onCollapsedClick( id );
-		} else {
-			// The task-related action
-			onClick();
-		}
+		this.isTaskCollapsed() ? this.props.onTaskTitleClick( id ) : onTaskTitleClick( 'none' );
```

Comment:
Let's use either `this.props.onTaskTitleClick` _or_ `onTaskTitleClick` consistently:

```suggestion
		this.isTaskCollapsed() ? onTaskTitleClick( id ) : onTaskTitleClick( 'none' );
```

Reply:
> Let's use either this.props.onTaskTitleClick or onTaskTitleClick consistently:

Can you elaborate on this a little more? Is it in this piece of code or elsewhere?

Reply:
Yes: of the two logical 'branches' of the ternary, one calls `this.props.onTaskTitleClick()`, while the other calls `onTaskTitleClick()`. However, both are the same thing -- the latter is being defined (via destructuring) on [R110](https://github.com/Automattic/wp-calypso/pull/33666/files#diff-acb49564d5ae2fcd2a24d573341a2d3dR110).

Thus, for consistency's sake, we should decide on one of them and use it in both cases. (Probably `onTaskTitleClick` since it wouldn't make much sense to destructure it otherwise.)

Reply:
Oh, I just realized that the method that we're calling this from has the exact same name :thinking: Okay, that could lead to some confusion.

Renaming the method to `expandTask` could provide an easy way out (see https://github.com/Automattic/wp-calypso/pull/33666/files#r292546081)

Reply:
Finally, I don't think we should use `'none'` as the magical value to indicate that the task is collapsed. We're already using `null` for the `initialState` [here](https://github.com/Automattic/wp-calypso/pull/33666/files#diff-109d16ece1aaa451541aafc75a84baf4R71).